lib/rack/request.rb

```diff
@@ -644,7 +648,7 @@ def parse_http_accept_header(header)
             quality = $1.to_f
           end
           [attribute, quality]
-        end
+        end.compact
```

Comment:
I think we should use a temporary variable and `map!` and `compact!` to avoid unnecessary array allocations.